core/src/main/java/org/commonjava/maven/ext/manip/impl/PluginManipulator.java

```diff
@@ -262,14 +295,45 @@ else if ( configPrecedence == Precedence.LOCAL )
                 {
                     logger.debug ("No remote configuration to inject from " + override.toString());
                 }
+                String oldVersion = plugin.getVersion();
                 // Always force the version in a pluginMgmt block or set the version if there is an existing
                 // one in build/plugins section.
-                if ( pluginMgmt || plugin.getVersion() != null )
+                if ( pluginMgmt || oldVersion != null )
                 {
-                    plugin.setVersion( override.getVersion() );
-                    logger.info( "Altered plugin version: " + groupIdArtifactId + "=" + override.getVersion());
-                }
+                    // Handle the situation where we are updating a dependency that has an existing property - in this
+                    // case we want to update the property instead.
+                    // TODO: Handle the scenario where the version might be ${....}${....}
+                    if ( oldVersion.startsWith( "${" ) )
```

Comment:
It looks like this is something we could move to PropertyUtils, maybe something like:

    public static String findPropertyRef( String version);

I'm just guessing if it's here we probably have similar logic somewhere else, related to dependencies?